sway/handlers.c

```diff
@@ -334,114 +331,109 @@ static bool handle_key(wlc_handle view, uint32_t time, const struct wlc_modifier
-	swayc_t *view = container_under_pointer();
-	if (pointer_state.floating.drag && view) {
-		if (view->is_floating) {
-			int dx = mouse_origin.x - prev_pos.x;
-			int dy = mouse_origin.y - prev_pos.y;
-			view->x += dx;
-			view->y += dy;
-			struct wlc_geometry geometry = {
-				.origin = {
-					.x = view->x,
-					.y = view->y
-				},
-				.size = {
-					.w = view->width,
-					.h = view->height
-				}
-			};
-			wlc_view_set_geometry(view->handle, 0, &geometry);
-			changed_floating = true;
-		} else {
-			swayc_t *init_view = pointer_state.tiling.init_view;
-			if (view != init_view && view->type == C_VIEW) {
-				changed_tiling = true;
-				int i, j;
-				for (i = 0; i < view->parent->children->length; i++) {
-					if (view->parent->children->items[i] == view) {
-						for (j = 0; j < init_view->parent->children->length; j++) {
-							if (init_view->parent->children->items[j] == init_view) {
-								double temp_w = view->width;
-								double temp_h = view->height;
-								view->width = init_view->width;
-								view->height = init_view->height;
-								init_view->width = temp_w;
-								init_view->height = temp_h;
-
-								init_view->parent->children->items[j] = view;
-								view->parent->children->items[i] = init_view;
-
-								swayc_t *temp = view->parent;
-								view->parent = init_view->parent;
-								init_view->parent = temp;
-
-								arrange_windows(&root_container, -1, -1);
-								break;
-							}
-						}
-						break;
-					}
-				}
-			}
-		}
-	} else if (pointer_state.floating.resize && view) {
-		changed_floating = resize_floating(prev_pos);
-	} else if (pointer_state.tiling.resize && view) {
-		changed_tiling = mouse_resize_tiled(prev_pos);
-	}
-	if (config->focus_follows_mouse && prev_handle != handle) {
-		// Dont change focus if fullscreen
-		swayc_t *focused = get_focused_view(view);
-		if (!swayc_is_fullscreen(focused)
-				&& !(pointer_state.l_held || pointer_state.r_held)) {
-			set_focused_container(container_under_pointer());
+	// Otherwise change focus if config is set an
+	else if (prev_view != pointer_state.view && config->focus_follows_mouse) {
+		if (pointer_state.view && pointer_state.view->type == C_VIEW) {
+			set_focused_container(pointer_state.view);
 		}
 	}
-	prev_handle = handle;
-	prev_pos = mouse_origin;
-	if (changed_tiling || changed_floating) {
-		return true;
-	}
 	return false;
 }
 
 
 static bool handle_pointer_button(wlc_handle view, uint32_t time, const struct wlc_modifiers *modifiers,
 		uint32_t button, enum wlc_button_state state, const struct wlc_origin *origin) {
-	swayc_t *focused = get_focused_container(&root_container);
-	// dont change focus if fullscreen
-	if (swayc_is_fullscreen(focused)) {
-		return false;
-	}
-	if (state == WLC_BUTTON_STATE_PRESSED) {
-		sway_log(L_DEBUG, "Mouse button %u pressed", button);
-		if (button == M_LEFT_CLICK) {
-			pointer_state.l_held = true;
+	enum { DONT_SEND_CLICK = true, SEND_CLICK = false };
```

Comment:
I don't like it. How about we have broader defines for `EVENT_PASSTHROUGH` and `EVENT_HANDLED` or something